<launch>
  <arg name="manager"             default="realsense_ros_camera_manager"/>
  <arg name="serial_no"           default="" />
  <arg name="camera"              default="camera"/>
  <arg name="depth"               default="depth" />
  <arg name="depth_width"         default="640" />
  <arg name="depth_height"        default="480" />
  <arg name="enable_depth"        default="true" />
  <arg name="depth_fps"           default="30" />
  <arg name="fisheye"             default="fisheye" />
  <arg name="fisheye_width"       default="640" />
  <arg name="fisheye_height"      default="480" />
  <arg name="enable_fisheye"      default="false" />
  <arg name="depth_fps"           default="30" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen"/> 
  <node pkg="nodelet" type="nodelet" name="realsense_ros_$(arg camera)" args="load realsense_ros_camera/NodeletDS5Camera $(arg manager)" output="screen">
    <param name="serial_no"                type="str"  value="$(arg serial_no)" />
    
    <param name="depth_width"              type="int"  value="$(arg depth_width)" />
    <param name="depth_height"             type="int"  value="$(arg depth_height)" />
    <param name="enable_depth"             type="bool" value="$(arg enable_depth)" />
    <param name="depth_fps"                type="int"  value="$(arg depth_fps)" />
    <param name="fisheye_width"            type="int"  value="$(arg fisheye_width)" />
    <param name="fisheye_height"           type="int"  value="$(arg fisheye_height)" />
    <param name="enable_fisheye"           type="bool" value="$(arg fisheye_depth)" />
    <param name="fisheye_fps"              type="int"  value="$(arg fisheye_fps)" />

    <param name="depth_optical_frame_id"   type="str"  value="$(arg camera)_depth_optical_frame" />
  </node>
</launch>
